### PR TITLE
Added a Component A and Component B

### PR DESCRIPTION
**Pull request by Sailanarmo**
Not sure if these are built correctly, I checked them both with the subnet manager driver and they both get responses. Need to check if they have been inserted into the routing table somehow.

### CODE REVIEW

**Review by LuxMiranda — CHANGES_REQUESTED** [review]
Looks good, but make sure to also copy over the .gitignore from Component_A into Component_B so that we don't get all of the CMakeFiles and binaries committed.

**Comment by LuxMiranda** [comment]
Awesome! Now that you have the new .gitignore, you can just go into the Component_B directory and do:
```
git rm -r . --cached
git add .
```
and then commit/push again

**Comment by LuxMiranda** [comment]
Make sure you also update the logical addresses. `la_CA` is the logical address for component A and `la_CB` is the one for component B.

File: demo/Component_B/component_B.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  ComponentB(std::shared_ptr<SpaCommunicator> com = nullptr) : Component(com) {}
+
+  virtual void handleSpaData(SpaMessage*) {}
+  virtual void sendSpaData(LogicalAddress) {}
+
+  virtual void appInit()
+  {
+    std::cout << "Component B initializing!" << '\n';
+
+    LocalHello hello(0, 0, la_LSM, la_CA, 0, 0, 0, 0);
```
Comment (LuxMiranda) [inline demo/Component_B/component_B.cpp]: This line should be:
`LocalHello hello(0, 0, la_LSM, la_CB, 0, 0, 0, 0);`

File: demo/Component_B/component_B.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    communicator->getLocalCommunicator()->clientConnect((SpaMessage*)&hello, sizeof(hello), messageCallback);
+
+    SubscriptionRequest request(la_CB, la_CA, la_LSM);
```
Comment (JackKiefer) [inline demo/Component_B/component_B.cpp]: Let's also scrap this in Component B for the moment and just get Component A subscribing to Component B

**Review by LuxMiranda — CHANGES_REQUESTED** [review]

File: demo/Component_B/component_B.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    std::cout << "Sending message with opcode: " << (int)request.spaMessage.spaHeader.opcode << "\n";
```
Comment (LuxMiranda) [inline demo/Component_B/component_B.cpp]: Let's also scrap this in Component B for the moment and just get Component A subscribing to Component B

**Review by LuxMiranda — CHANGES_REQUESTED** [review]

**Review by LuxMiranda — APPROVED** [review]

**Pull request merged**